cpp/src/community/extract_subgraph_by_vertex.cu

```diff
@@ -0,0 +1,167 @@
+                                    vertex_t num_vertices,
+                                    cugraph::experimental::GraphCOO<vertex_t, edge_t, weight_t> &result,
+                                    cudaStream_t stream) {
+
+    edge_t    graph_num_verts = graph.number_of_vertices;
+
+    rmm::device_vector<int64_t> error_count_v{1, 0};
+    rmm::device_vector<vertex_t> vertex_used_v{graph_num_verts, num_vertices};
+
+    vertex_t *d_vertex_used = vertex_used_v.data().get();
+    int64_t  *d_error_count = error_count_v.data().get();
+
+    thrust::for_each(rmm::exec_policy(stream)->on(stream),
+                     thrust::make_counting_iterator<vertex_t>(0),
+                     thrust::make_counting_iterator<vertex_t>(num_vertices),
+                     [vertices, d_vertex_used, d_error_count, graph_num_verts]
+                     __device__ (vertex_t idx) {
+                       vertex_t v = vertices[idx];
+                       if ((v >= 0) && (v < graph_num_verts)) {
+                         d_vertex_used[v] = idx;
+                       } else {
+                         cugraph::atomicAdd(d_error_count, int64_t{1});
+                       }
+                     });
+
+    CUGRAPH_EXPECTS(error_count_v[0] == 0, "Input error... vertices specifies vertex id out of range");
+
+    vertex_t *graph_src = graph.src_indices;
+    vertex_t *graph_dst = graph.dst_indices;
+    weight_t *graph_weight = graph.edge_data;
+
+    // iterate over the edges and count how many make it into the output
+    int64_t count = thrust::count_if(rmm::exec_policy(stream)->on(stream),
+                                     thrust::make_counting_iterator<edge_t>(0),
+                                     thrust::make_counting_iterator<edge_t>(graph.number_of_edges),
+                                     [graph_src, graph_dst, d_vertex_used, num_vertices]
+                                     __device__ (edge_t e) {
+                                       vertex_t s = graph_src[e];
+                                       vertex_t d = graph_dst[e];
+                                       return ((d_vertex_used[s] < num_vertices) && (d_vertex_used[d] < num_vertices));
+                                     });
+
+    if (count > 0) {
+#if 0
+      rmm::device_vector<vertex_t> new_src_v(count);
+      rmm::device_vector<vertex_t> new_dst_v(count);
+      rmm::device_vector<weight_t> new_weight_v;
+
+      vertex_t *d_new_src = new_src_v.data().get();
+      vertex_t *d_new_dst = new_dst_v.data().get();
+      weight_t *d_new_weight{nullptr};
+
+      if (has_weight) {
+        new_weight_v.resize(count);
+        d_new_weight = new_weight_v.data().get();
+      }
+#endif
+      vertex_t *d_new_src{nullptr};
+      vertex_t *d_new_dst{nullptr};
+      weight_t *d_new_weight{nullptr};
+
+      ALLOC_TRY(&d_new_src, count * sizeof(vertex_t), nullptr);
+      ALLOC_TRY(&d_new_dst, count * sizeof(vertex_t), nullptr);
```

Comment:
We need to remove ALLOC_TRY as well, ALLOC_TRY calls RMM_ALLOC.

Reply:
My comment above was intended to address this.  This is temporary code until #799 is merged.  At that point the input will be a GraphCOOView and the output will be a GraphCOO, so these will have been allocated as rmm::device_buffer objects.